packages/blockchain-extension/extension/commands/environmentConnectCommand.ts

```diff
@@ -56,6 +56,24 @@ export async function fabricEnvironmentConnect(fabricEnvironmentRegistryEntry: F
             fabricEnvironment = new FabricEnvironment(fabricEnvironmentRegistryEntry.name);
         }
 
+        let nodes: FabricNode[] = await fabricEnvironment.getNodes();
+
+        if (fabricEnvironmentRegistryEntry.url) {
+            let informOfChanges: boolean = true;
+            if (nodes.length === 0) {
+                const importNodes: boolean = await UserInputUtil.showConfirmationWarningMessage(`Error connecting to environment ${fabricEnvironmentRegistryEntry.name}: no visible nodes. Would you like to filter nodes?`);
+                if (!importNodes) {
+                    return;
+                }
+                informOfChanges = false;
+            }
+            await vscode.commands.executeCommand(ExtensionCommands.EDIT_NODE_FILTERS, fabricEnvironmentRegistryEntry, false, UserInputUtil.ADD_ENVIRONMENT_FROM_OPS_TOOLS, informOfChanges);
+            nodes = await fabricEnvironment.getNodes();
+            if (nodes.length === 0) {
+                return;
```

Comment:
Not at this point, because this PR deals with reconnect, so we are coming from a disconnected scenario. Next PR will deal with refresh, ie, connect when already connected. This will be addressed then.